### PR TITLE
⬆ Upgrade version required of Starlette from `0.19.1` to `0.20.4`

### PR DESCRIPTION
**Pull request by Kludex**
Runs against Starlette's ~master branch~ 0.20.0.

This version drops Python 3.6.

---

Edit by @tiangolo:

Upgrade Starlette version to `0.20.4`, update imports for `ExceptionMiddleware` from the new location in Starlette.

Starlette is updated directly to `0.20.4` here as those 4 patch versions had several bug fixes, including security ones, this way it's possible to skip that, upgrade to the next feature version (from 0.19 to 0.20) and get all the recent bug fixes.

Support for Python 3.6 was just dropped in a previous PR. This PR also includes removing Pytest markers that were only needed for Python 3.6.

### CODE REVIEW

**Review by anukaal — APPROVED** [review]

**Review by BilalAlpaslan — APPROVED** [review]

File: pyproject.toml [hunk 1]
```diff
@@ -35,11 +34,11 @@ classifiers = [
     "Topic :: Internet :: WWW/HTTP",
 ]
 requires = [
-    "starlette ==0.17.1",
+    "starlette @ git+https://github.com/encode/starlette.git#master",
```
Comment (Kludex) [inline pyproject.toml:37]:
```suggestion
    "starlette ==0.20.0",
```

File: tests/test_union_inherited_body.py [hunk 2]
```diff
@@ -4,8 +4,6 @@
 from fastapi.testclient import TestClient
 from pydantic import BaseModel
 
-from .utils import needs_py37
-
 # In Python 3.6:
```
Comment (musicinmybrain) [inline tests/test_union_inherited_body.py:7]: It seems like this comment about differences between Python 3.6 and 3.7 can be removed, too.

File: fastapi/exceptions.py [hunk 3]
```diff
@@ -1,4 +1,4 @@
-from typing import Any, Dict, Optional, Sequence, Type
+from typing import Any, Sequence, Type
```
Comment (Kludex) [inline fastapi/exceptions.py:1]:
```suggestion
from typing import Any, Dict, Optional, Sequence, Type
```

File: pyproject.toml [hunk 4]
```diff
@@ -35,11 +34,11 @@ classifiers = [
     "Topic :: Internet :: WWW/HTTP",
 ]
 requires = [
-    "starlette==0.19.1",
+    "starlette ==0.20.0",
```
Comment (Kludex) [inline pyproject.toml:37]:
```suggestion
    "starlette==0.20.0",
```

**Comment by MatthijsKok** [comment]
0.84.0 was just released. It drops support for Python 3.6, which this PR required.  
Does that mean this branch can now be rebased and reviewed?  
I can contribute to anything if needed, incrementally bumping starlette to its newest version will include some nice bugfixes.

**Comment by tiangolo** [comment]
Thanks @Kludex! :rocket: 

I updated it a bit, raised the version to `0.20.4` and added a couple of tweaks.

Thanks for the discussion and help here everyone! :coffee:

**Pull request merged**